### PR TITLE
feat: add stopwatch package

### PR DESCRIPTION
**Pull request by Soumil07**

### CODE REVIEW

File: packages/stopwatch/package.json [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	"main": "dist/index.js",
+	"module": "dist/index.mjs",
+	"browser": "dist/index.umd.js",
+	"unpkg": "dist/index.umd.js",
+	"types": "dist/index.d.ts",
+	"exports": {
+		"import": "./dist/index.mjs",
+		"require": "./dist/index.js"
+	},
+	"homepage": "https://github.com/sapphire-project/utilities/tree/main/packages/stopwatcj",
```
Comment (Favna) [inline packages/stopwatch/package.json:16]:
```suggestion
	"homepage": "https://github.com/sapphire-project/utilities/tree/main/packages/stopwatch",
```

File: packages/stopwatch/package.json [hunk 2]
```diff
@@ -0,0 +1,52 @@
+{
+	"name": "@sapphire/stopwatch",
+	"version": "1.0.0",
+	"description": "A simple Stopwatch class for JavaScript.",
```
Comment (Favna) [inline packages/stopwatch/package.json:4]:
```suggestion
	"description": "Accurately measure passing time.",
```

File: packages/stopwatch/package.json [hunk 3]
```diff
@@ -0,0 +1,52 @@
+{
+	"name": "@sapphire/stopwatch",
+	"version": "1.0.0",
```
Comment (Favna) [inline packages/stopwatch/package.json:3]:
```suggestion
	"version": "0.0.0",
```

Version bump to 1.0.0 will be done by Lerna on first publish

File: packages/stopwatch/package.json [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+	"description": "A simple Stopwatch class for JavaScript.",
+	"author": "@sapphire",
+	"license": "MIT",
+	"main": "dist/index.js",
+	"module": "dist/index.mjs",
+	"browser": "dist/index.umd.js",
+	"unpkg": "dist/index.umd.js",
```
Comment (Favna) [inline packages/stopwatch/package.json:10]:
```suggestion
```

File: packages/stopwatch/package.json [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+	"types": "dist/index.d.ts",
+	"exports": {
+		"import": "./dist/index.mjs",
+		"require": "./dist/index.js"
+	},
+	"homepage": "https://github.com/sapphire-project/utilities/tree/main/packages/stopwatcj",
+	"scripts": {
+		"test": "jest",
+		"lint": "eslint src tests --ext ts --fix -c ../../.eslintrc",
+		"build": "tsc -p .",
```
Comment (Favna) [inline packages/stopwatch/package.json:20]:
```suggestion
		"build": "tsc -b src",
		"postbuild": "gen-esm-wrapper dist/index.js dist/index.mjs",
```

Also add a root dev dependency for `gen-esm-wrapper`

File: packages/stopwatch/tests/chainables.test.ts [hunk 6]
```diff
@@ -0,0 +1,57 @@
+import { sleep } from './shared';
+import { Stopwatch } from '../src';
+
```
Comment (Favna) [inline packages/stopwatch/tests/chainables.test.ts:3]: Wrap the entire thing in 
```ts
describe('Chainable', () => {
  // tests here
});
```

File: packages/stopwatch/tests/duration.test.ts [hunk 7]
```diff
@@ -0,0 +1,14 @@
+import { sleep } from './shared';
+import { Stopwatch } from '../src';
+
```
Comment (Favna) [inline packages/stopwatch/tests/duration.test.ts:3]: Wrap the entire thing in 
```ts
describe('Duration', () => {
  // tests here
});
```

File: packages/stopwatch/tests/running.test.ts [hunk 8]
```diff
@@ -0,0 +1,19 @@
+import { Stopwatch } from '../src';
+
```
Comment (Favna) [inline packages/stopwatch/tests/running.test.ts:2]: Wrap the entire thing in 
```ts
describe('Running', () => {
  // tests here
});
```

File: packages/stopwatch/tests/stop-accuracy.test.ts [hunk 9]
```diff
@@ -0,0 +1,19 @@
+import { sleep } from './shared';
+import { Stopwatch } from '../src';
+
```
Comment (Favna) [inline packages/stopwatch/tests/stop-accuracy.test.ts:3]: Wrap the entire thing in 
```ts
describe('Stop Accuracy', () => {
  // tests here
});
```

File: packages/stopwatch/tests/toString.test.ts [hunk 10]
```diff
@@ -0,0 +1,26 @@
+import { sleep } from './shared';
+import { Stopwatch } from '../src';
+
```
Comment (Favna) [inline packages/stopwatch/tests/toString.test.ts:3]: Wrap the entire thing in 
```ts
describe('ToString', () => {
  // tests here
});
```

File: packages/stopwatch/tsconfig.json [hunk 11]
```diff
@@ -0,0 +1,10 @@
+{
+	"extends": "../ts-config/src/tsconfig.json",
+	"compilerOptions": {
+		"target": "ES2020",
+		"module": "ESNext",
+		"outDir": "build",
+		"rootDir": "src"
```
Comment (Favna) [inline packages/stopwatch/tsconfig.json:7]:
```suggestion
		"outDir": "build",
		"rootDir": "src",
		"composite": true,
		"tsBuildInfoFile": "./dist/.tsbuildinfo"
```

This has to build as CommonJS

File: packages/stopwatch/tests/tsconfig.json [hunk 12]
```diff
@@ -0,0 +1,11 @@
+{
+	"extends": "../tsconfig.json",
+	"compilerOptions": {
+		"target": "ES2020",
```
Comment (Favna) [inline packages/stopwatch/tests/tsconfig.json:4]:
```suggestion
```

**Review by Favna — CHANGES_REQUESTED** [review]
Also missing `src/tsconfig.json`, see decorators package for example (though don't include the `importsNotUsedAsValues` in that, that shouldn't even be there)

And also add `packages/stopwatch/src/index.ts` to `typedoc.json`

File: packages/stopwatch/package.json [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+	"types": "dist/index.d.ts",
+	"exports": {
+		"import": "./dist/index.mjs",
+		"require": "./dist/index.js"
+	},
+	"homepage": "https://github.com/sapphire-project/utilities/tree/main/packages/stopwatch",
+	"scripts": {
+		"test": "jest",
+		"lint": "eslint src tests --ext ts --fix -c ../../.eslintrc",
+		"build": "tsc -p src",
```
Comment (Favna) [inline packages/stopwatch/package.json:18]:
```suggestion
		"build": "tsc -b src",
```

-b is build targets
-p is project

While both do work in this case, -b is what we use elsewhere.

**Review by Favna — CHANGES_REQUESTED** [review]
One last thing

**Review by Favna — DISMISSED** [review]

**Review by kyranet — APPROVED** [review]

File: packages/stopwatch/tests/chainables.test.ts [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+		await sleep(1000);
+		stopwatch.stop();
+		expect(stopwatch.duration >= 990 && stopwatch.duration <= 1200).toBe(true);
+
+		stopwatch.reset();
+
+		expect(stopwatch.duration === 0).toBe(true);
+	});
+
+	test('start(stopped stopwatch)', async () => {
```
Comment (vladfrangu) [inline packages/stopwatch/tests/chainables.test.ts:31]:
```suggestion
	test('start stopped stopwatch)', async () => {
```

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: packages/stopwatch/tests/chainables.test.ts [hunk 15]
```diff
@@ -0,0 +1,59 @@
...
+		expect(stopwatch.duration >= 990 && stopwatch.duration <= 1200).toBe(true);
+
+		stopwatch.start();
+
+		await sleep(1000);
+		stopwatch.stop();
+		expect(stopwatch.duration >= 1990 && stopwatch.duration <= 2200).toBe(true);
+	});
+
+	test('stopping(twice)', async () => {
```
Comment (Favna) [inline packages/stopwatch/tests/chainables.test.ts:46]:
```suggestion
	test('stopping the same stopwatch twice', async () => {
```

File: packages/stopwatch/tests/chainables.test.ts [hunk 16]
```diff
@@ -0,0 +1,59 @@
...
+		await sleep(1000);
+		stopwatch.stop();
+		expect(stopwatch.duration >= 990 && stopwatch.duration <= 1200).toBe(true);
+
+		stopwatch.reset();
+
+		expect(stopwatch.duration === 0).toBe(true);
+	});
+
+	test('start stopped stopwatch)', async () => {
```
Comment (Favna) [inline packages/stopwatch/tests/chainables.test.ts:31]:
```suggestion
	test('starting a previously stopped stopwatch', async () => {
```

**Review by Favna — APPROVED** [review]

**Review by vladfrangu — APPROVED** [review]

**Pull request merged**